oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/logout/LogoutTokenClaimAccessor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.oidc.authentication.logout;
+
+import java.net.URL;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.security.oauth2.core.ClaimAccessor;
+import org.springframework.security.oauth2.jwt.JwtClaimAccessor;
+
+/**
+ * A {@link ClaimAccessor} for the &quot;claims&quot; that can be returned in OIDC
+ * Backchannel Logout Tokens
+ *
+ * @author Josh Cummings
+ * @since 6.1
+ * @see OidcLogoutToken
+ * @see <a target="_blank" href=
+ * "https://openid.net/specs/openid-connect-backchannel-1_0.html#LogoutToken">Logout
+ * Token</a>
+ */
+public interface LogoutTokenClaimAccessor extends JwtClaimAccessor {
```

Comment:
https://openid.net/specs/openid-connect-backchannel-1_0.html#Terminology defines the logout token as a JWT. What is the rationale for using `ClaimAccessor` instead?

Reply:
It aligns with `IdTokenClaimAccessor`

Reply:
While helpful, that generates another question: given that ID tokens are JWTs, why doesn't `IdTokenClaimAccessor` extend `JwtClaimAccessor`? It seems like `IdTokenClaimAccessor` could do the following:

```java
public class IdTokenClaimAccessor extends StandardClaimAccessor, JwtClaimAccessor
```

to the removal of several duplicate methods.

And while I realize that changing `IdTokenClaimAccessor` isn't part of this PR, I wonder if instead, we should add a ticket for enhancing `IdTokenClaimAccessor` and `LogoutTokenClaimAccessor` to both extend `JwtClaimAccessor` since the corresponding tokens are JWTs. What do you think?

And then double-checking here, `IdTokenClaimAccessor` appears to extend `StandardClaimAccessor`. To align with `IdTokenClaimAccessor`, should `LogoutTokenClaimAccessor` extend `StandardClaimAccessor`?

Reply:
I believe the reason why we didn't go with `IdTokenClaimAccessor extends JwtClaimAccessor` is because we would introduce a required dependency of `oauth2-client` -> `oauth2-jose`, which isn't ideal as `oauth2-client` should only depend on `oauth2-core` and optionally `oauth2-jose` if JWT's are being used.

Furthermore, if you look at `OidcIdToken` it doesn't truly represent a JWT since there is no `headers` attribute. It only contains a `claims` attribute, hence the reason for `IdTokenClaimAccessor` for accessing the claims only.

Reply:
Okay, thanks for the background. Can you address this question as well, please?

> And then double-checking here, `IdTokenClaimAccessor` appears to extend `StandardClaimAccessor`. To align with `IdTokenClaimAccessor`, should `LogoutTokenClaimAccessor` extend `StandardClaimAccessor`?

Reply:
> should `LogoutTokenClaimAccessor` extend `StandardClaimAccessor`?

This would not make sense since the Logout Token does not contain any of the [standard claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims). Those only exist in the ID Token.